src/passwordPenalty.js

```diff
@@ -5,10 +5,12 @@
  * @param {string} password
  * @returns {number}
  */
-export default function penaltyPoints(password = "") {
+function penaltyPoints(password = "") {
+  if (password === null) return 0;
   // The following line ensures, that password is always a string, like the number 128 -> string "128"
   if (typeof password !== "string") password = String(password);
-  let groupedChar = password.match(/([a-zA-Z0-9])\1+/g);
+  //the line is about grouping repeated char separatedly into an array
+  const groupedChar = password.match(/([a-zA-Z0-9])\1+/g);
   if (!groupedChar) return 0;
   let count = 0;
   groupedChar.map((elt) => {
```

Comment:
instead of using map which create new array, you can use other methods like forEach and reduce this would help a lot